### PR TITLE
fix(backup): modify backup volume/volume APIs and naming

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#5411

### CODE REVIEW

File: api/backup.go [hunk 1]
```diff
@@ -342,22 +342,22 @@ func (s *Server) BackupDelete(w http.ResponseWriter, req *http.Request) error {
 		return errors.New("empty backup name is not allowed")
 	}
 
-	volName := mux.Vars(req)["volName"]
+	backupVolumeName := mux.Vars(req)["name"]
```
Comment (derekbit) [inline api/backup.go:345]: I prefer
```suggestion
	backupVolumeName := mux.Vars(req)["backupVolumeName"]
```

WDYT?

File: api/backup.go [hunk 2]
```diff
@@ -315,14 +315,14 @@ func (s *Server) BackupGet(w http.ResponseWriter, req *http.Request) error {
 	if input.Name == "" {
 		return errors.New("empty backup name is not allowed")
 	}
-	volName := mux.Vars(req)["volName"]
+	backupVolumeName := mux.Vars(req)["name"]
```
Comment (derekbit) [inline api/backup.go:318]: Prefer
```suggestion
	backupVolumeName := mux.Vars(req)["backupVolumeName"]
```

File: csi/controller_server.go [hunk 3]
```diff
@@ -280,11 +282,12 @@ func (cs *ControllerServer) getBackupVolume(volumeName string) (*longhornclient.
...
 	}
 
-	if len(list.Data) >= 2 {
+	if len(list.Data) > 1 {
 		return nil, fmt.Errorf("found multiple backup volumes for backup target %s and volume %s", vol.BackupTargetName, volumeName)
 	}
 	if len(list.Data) == 0 {
-		return nil, fmt.Errorf("failed to find backup volume for backup target %s and volume %s", vol.BackupTargetName, volumeName)
+		log.Debugf("cannot find backup volume with backup target %s and volume %s", vol.BackupTargetName, volumeName)
+		return cs.apiClient.BackupVolume.ById("non-existing-" + utils.RandomID(8))
```
Comment (derekbit) [inline csi/controller_server.go:290]: What's the purpose of the change?

File: csi/controller_server.go [hunk 4]
```diff
@@ -1245,7 +1248,7 @@ func (cs *ControllerServer) waitForBackupControllerSync(volumeName, snapshotName
 	if err != nil {
 		return nil, err
 	}
-	if backup.SnapshotCreated != "" {
+	if backup != nil && backup.SnapshotCreated != "" {
```
Comment (derekbit) [inline csi/controller_server.go:1251]: Why can `cs.getBackup` return `nil error` and `nil backup` simutaneously?

File: api/backup.go [hunk 5]
```diff
@@ -265,9 +265,9 @@ func (s *Server) SyncBackupVolume(w http.ResponseWriter, req *http.Request) erro
 }
 
 func (s *Server) BackupVolumeDelete(w http.ResponseWriter, req *http.Request) error {
-	volName := mux.Vars(req)["volName"]
-	if err := s.m.DeleteBackupVolume(volName); err != nil {
-		return errors.Wrapf(err, "failed to delete backup volume '%s'", volName)
+	backupVolumeName := mux.Vars(req)["name"]
```
[the remark this answers is not in the document]
Comment (mantissahz) [reply api/backup.go]: Modified. Thanks

File: manager/kubernetes.go [hunk 6]
```diff
@@ -35,13 +36,29 @@ func (m *VolumeManager) PVCreate(name, pvName, fsType, secretNamespace, secretNa
...
+		}
+		if backup != nil {
+			backupTargetName = backup.Status.BackupTargetName
+			if backupTargetName == "" {
+				backupTargetName = backup.Labels[types.LonghornLabelBackupTarget]
+			}
+			if backupTargetName == "" {
+				return nil, fmt.Errorf("failed to get backup target name for backup %v", bName)
 			}
 		}
```
[the remark this answers is not in the document]
Comment (mantissahz) [reply manager/kubernetes.go]: Modified. Thanks

File: csi/controller_server.go (hunk 3 above)
Comment (mantissahz) [reply csi/controller_server.go]: Modified as discussed. Thanks

File: csi/controller_server.go (hunk 4 above)
Comment (mantissahz) [reply csi/controller_server.go:1250]: Modified as discussed. Thanks

File: csi/controller_server.go [hunk 7]
```diff
@@ -128,7 +128,7 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 					}
 					backupVolume, backupName := sourceVolumeName, id
 					bv, err := cs.getBackupVolume(backupVolume)
-					if err != nil {
+					if bv == nil || err != nil {
```
[the remark this answers is not in the document]
Comment (derekbit) [reply csi/controller_server.go:131]: @mantissahz Can you update the error message as coderabbitai suggested?

[the remark this answers is not in the document]
Comment (mantissahz) [reply csi/controller_server.go]: good suggestion. Modified. Thanks.

**Review by derekbit — APPROVED** [review]
